torch/_library/infer_schema.py

```diff
@@ -318,3 +319,149 @@ def tuple_to_list(tuple_type: type[tuple]) -> type[list]:
         return list[type_args[0]]  # type: ignore[valid-type]
     else:
         return list[typing.Union[tuple(type_args)]]  # type: ignore[misc, return-value]
+
+
+# Below is an implementation of generating FunctionSchema from example values.
```

Comment:
Can you toss this into a new file? infer_schema is for inferring the schema from a function. I think you just want a hop_schema file.

Maybe even toss it under torch._higher_order_ops.schema

Reply:
Sure, torch._higher_order_ops.schema sounds good.